core/audits/bf-cache.js

```diff
@@ -0,0 +1,123 @@
+   */
+  displayValue: `{itemCount, plural,
+    =1 {1 actionable failure reason}
+    other {# actionable failure reasons}
+    }`,
+};
+/* eslint-enable max-len */
+
+const str_ = i18n.createIcuMessageFn(import.meta.url, UIStrings);
+
+/** @type {LH.Crdp.Page.BackForwardCacheNotRestoredReasonType[]} */
+const ORDERED_FAILURE_TYPES = ['PageSupportNeeded', 'Circumstantial', 'SupportPending'];
+
+/** @type {Record<LH.Crdp.Page.BackForwardCacheNotRestoredReasonType, string | LH.IcuMessage>} */
+const FAILURE_TYPE_TO_STRING = {
+  PageSupportNeeded: str_(UIStrings.actionableFailureType),
+  Circumstantial: str_(UIStrings.notActionableFailureType),
+  SupportPending: str_(UIStrings.supportPendingFailureType),
+};
+
+class BFCache extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'bf-cache',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      supportedModes: ['navigation', 'timespan'],
+      requiredArtifacts: ['BFCacheFailures'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts) {
+    const failures = artifacts.BFCacheFailures;
+    if (!failures.length) return {score: 1};
+
+    // TODO: Analyze more than one bf cache failure.
+    const {notRestoredReasonsTree} = failures[0];
+
+    /** @type {LH.Audit.Details.TableItem[]} */
+    const results = [];
+    let numActionable = 0;
+
+    for (const failureType of ORDERED_FAILURE_TYPES) {
+      const reasonsMap = notRestoredReasonsTree[failureType];
+
+      for (const [reason, frameUrls] of Object.entries(reasonsMap)) {
+        if (failureType === 'PageSupportNeeded') numActionable++;
+
+        results.push({
+          reason: NotRestoredReasonDescription[reason]?.name ?? reason,
+          failureType: FAILURE_TYPE_TO_STRING[failureType],
+          subItems: {
+            type: 'subitems',
+            items: frameUrls.map(frameUrl => ({frameUrl})),
+          },
+          // Include hidden protocol reason code for debugging.
+          protocolReason: reason,
+        });
+      }
+    }
+
+    /** @type {LH.Audit.Details.Table['headings']} */
+    const headings = [
+      /* eslint-disable max-len */
+      {key: 'reason', valueType: 'text', subItemsHeading: {key: 'frameUrl', valueType: 'url'}, label: str_(UIStrings.failureReasonColumn)},
+      {key: 'failureType', valueType: 'text', label: str_(UIStrings.failureTypeColumn)},
+      /* eslint-enable max-len */
+    ];
+
+    const details = Audit.makeTableDetails(headings, results);
+
+    const displayValue = numActionable ?
+      str_(UIStrings.displayValue, {itemCount: numActionable}) :
+      undefined;
+
+    return {
+      score: numActionable ? 0 : 1,
```

Comment:
Oh, I actually missed this. This shouldn't pass even if it's not actionable...it sucks for the user, but we really don't want to give the impression that your page is good for bfcache (at least on load) when it's not.

Lots of failing audits is also good motivation to fix non-actionable reasons or at least transform them into actionable ones.

Reply:
Another option would be to issue an audit-level warning if there are only non-actionable reasons.

Additionally, if we go with your suggestion we should also consider the other suggestions for the audit title.

Reply:
This is what that could look like:

<img width="781" alt="Screen Shot 2022-12-14 at 12 39 47 PM" src="https://user-images.githubusercontent.com/6752989/207709119-242575a0-8bf1-4d09-b774-9f7deccce06b.png">

Unfortunately the user still has to expand the passed audit dropdown to see it. Maybe we could add a "Passed with warning" section to performance like other categories?

Reply:
I think it should fail. It didn’t pass - it doesn’t matter that it was for an inactionable reason.